AWSKinesis/AWSAbstractKinesisRecorder.m

```diff
@@ -178,17 +178,14 @@ - (AWSTask *)saveRecord:(NSData *)data
                                ];
                 if (!result) {
                     AWSLogError(@"SQLite error. [%@]", db.lastError);
-                    error = db.lastError;
+                    dbError = db.lastError;
                 }
             }];
         }
-
-        if (error) {
```

Comment:
Shall we really return here? 
I thought the code bellow shouldn't be locked by the errors from the code above